### ISSUE TITLE
Evaluation batch inference with same image size

### ISSUE THREAD

**Issue by mmeendez8** [open-mmlab/mmsegmentation#2965]
I understand the problematic of batch inference with different image sizes. But we should be able to do this on a dataset with constant image size. Is there any plan of supporting this feature?

**Comment by irabanillo91** [comment]
I agree, I find it quite limiting not being able to run inference on batches since it slows down a lot the processing in production. Would it be possible to remove this assert and let the user deal with it?

**Comment by joihn** [comment]
l also feel like this is a big limitation. My workflow involve a validation epoch after each train epoch, and with `batch_size=1` it is quite slow
